internal/rpmmd/repository.go

```diff
@@ -212,7 +212,15 @@ type RPMMD interface {
 	// Depsolve takes a list of required content (specs), explicitly unwanted content (excludeSpecs), list
 	// or repositories, and platform ID for modularity. It returns a list of all packages (with solved
 	// dependencies) that will be installed into the system.
+	//
+	// NOTE: Kept for legacy purposes (Weldr). New code should default to using DepsolvePackageSets().
```

Comment:
I think that we should use `Deprecated:` instead of `NOTE:`, see https://github.com/golang/go/wiki/Deprecated

This will probably lead to the linter blowing up in Weldr so we will need to add some ignore lines but if will very effectively prevent people from using the old function.

Reply:
I was not 100% if we really want to deprecate this method, but your suggestion makes sense.